docs/decisions/0010-limit-modulestore-use-in-lms.rst

```diff
@@ -0,0 +1,273 @@
+  setup and running of tests. Modulestore has proven to be much more challenging
+  to extract in this way, and any external apps relying on the modulestore will
+  be forced to use dependency inversion mechanisms that are more vulnerable to
+  breaking as edx-platform evolves.
+
+
+Conversion Guide
+=================
+
+There are a couple of performant alternatives to the Modulestore for common use
+cases. If neither of these alternatives satisfies your use case, please see the
+"Advanced Use Cases" section.
+
+Course Settings
+***************
+
+Apps that only query the Modulestore for course settings that are stored on the
+root CourseBlock should be modified to query CourseOverviews instead. This is
+available from ``openedx.core.djangoapps.content.course_overview.api``, which
+has the ``get_course_overview`` and ``get_course_overviews`` functions.
+
+If the settings data is not currently captured in the CourseOverview model, do
+the following:
+
+1. Add the fields to the CourseOverview model, with default values.
+2. Generate the migration file.
+3. Increment CourseOverview.VERSION by one.
+4. Update CourseOverview._create_or_update to properly load the data from the
+   CourseBlock object (from modulestore) and into the CourseOverview class.
+   Attributes are typically named the same on both classes.
+
+This will cause a performance penalty on initial rollout. The CourseOverviews
+API will make sure that you don't get an old version of the data, and will force
+just-in-time re-generation of the overview if the stored record version is less
+than CourseOverview.VERSION. This should resolve a few minutes after deployment.
+
+Course Outline / Sequence Metadata
+**********************************
+
+Apps that need course outline data and metadata about sequences can make use
+of the Learning Sequences public API from the
+``openedx.core.djangoapps.content.learning_sequences.api`` package. A couple of
+important caveats to the functions advertised here:
+
+* Old Mongo courses are not supported. These are soon-to-be-removed courses with
+  course keys of the format "Org/Course/Run". All course keys that begin with
+  "course-v1:" or "ccx-v1:" are supported.
+* This is a new API, and new API functions and data will be added over time.
+
+Advanced Use Cases
+******************
+
+Sometimes the data an application needs just isn't available in a performant way
+in the LMS. When this happens, you'll need to create your own model for this
+data in your application, and then push new data to it during the course publish
+process. To demonstrate how this can work, let's walk through the implementation
+of Learning Sequences, which follows this general architectural pattern.
+
+
+Studio Process
+^^^^^^^^^^^^^^
+
+The code to extract data from the Modulestore and convert it into a format that
+the Learning Sequences app understands lives in the ``cms.djangoapps.contentstore``
+package. This has several pieces:
+
+Data Extraction
```

Comment:
The rendered RST doesn't make it clear that these are sub-headings under Studio process. Will it let you do another level of heading nesting?